homeassistant/components/nws/weather.py

```diff
@@ -165,85 +73,73 @@ def convert_condition(time, weather):
     return cond, max(prec_probs)
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info):
     """Set up the NWS weather platform."""
-
     latitude = config.get(CONF_LATITUDE, hass.config.latitude)
     longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
     station = config.get(CONF_STATION)
-    api_key = config[CONF_API_KEY]
-    mode = config[CONF_MODE]
 
-    websession = async_get_clientsession(hass)
-    # ID request as being from HA, pynws prepends the api_key in addition
-    api_key_ha = f"{api_key} homeassistant"
-    nws = SimpleNWS(latitude, longitude, api_key_ha, websession)
+    nws_data = hass.data[DOMAIN][base_unique_id(latitude, longitude)]
 
-    _LOGGER.debug("Setting up station: %s", station)
     try:
-        await nws.set_station(station)
-    except ERRORS as status:
-        _LOGGER.error(
-            "Error getting station list for %s: %s", (latitude, longitude), status
-        )
+        await nws_data.async_set_station(station)
+    except (aiohttp.ClientError, asyncio.TimeoutError) as err:
+        _LOGGER.error("Error automatically setting station: %s", str(err))
         raise PlatformNotReady
 
-    _LOGGER.debug("Station list: %s", nws.stations)
-    _LOGGER.debug(
-        "Initialized for coordinates %s, %s -> station %s",
-        latitude,
-        longitude,
-        nws.station,
-    )
+    await nws_data.async_update()
 
-    async_add_entities([NWSWeather(nws, mode, hass.config.units, config)], True)
+    async_add_entities(
+        [
+            NWSWeather(nws_data, "daynight", hass.config.units),
```

Comment:
Ideally the PyPi package provides these constants.  No need to change now though -- there are a lot here so good for a followup
```suggestion
            NWSWeather(nws_data, SOME_CONSTANT_FOR_DAYLIGHT, hass.config.units),
```

Reply:
This is a home assistant specific thing IMO.  The upstream pypi library just provides methods and attributes to retrieve the hourly and day+night forecasts.  Each entity needs to know which forecast it is getting, which is why I think this is a home assistant feature.  I should make these true constants either in this module or const though.

Reply:
Probably a good thing to do in a followup since you have the config flow coming